### ISSUE TITLE
The version of pip installed by dh_virtualenv in the app venv is erroring out

### ISSUE THREAD

**Issue by zenmonkeykstop** [freedomofpress/securedrop#5940]
## Description

As part of release prep, the securedrop builder image was updated, pulling in a new Ubuntu `python3-pip` package in the process. This package is installed in the `securedrop-app-code` venv by dh_virtualenv during the build process (whether it should be is an open question), and is used in a `testinfra` test that checks expected package install versions against the versions reported by pip.

## Steps to Reproduce

- Update `molecule/builder-focal/create.yml`, setting `default_image` to `quay.io/freedomofpress/sd-docker-builder-focal:2021_05_13`
- run `make build-debs && make staging && make testinfra`

## Expected Behavior
build succeeds, staging set up, testinfra passes

## Actual Behavior
build succeeds, staging set up, testinfra fails on `test_app_pip_deps` with traceback:

```
Traceback (most recent call last):
  File "/opt/venvs/securedrop-app-code/bin/pip", line 5, in <module>
    from pip._internal.cli.main import main
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/pip/_internal/cli/main.py", line 10, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/pip/_internal/cli/autocompletion.py", line 9, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/pip/_internal/cli/main_parser.py", line 7, in <module>
    from pip._internal.cli import cmdoptions
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/pip/_internal/cli/cmdoptions.py", line 25, in <module>
    from pip._internal.locations import USER_CACHE_DIR, get_src_prefix
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/pip/_internal/locations.py", line 19, in <module>
    from pip._internal.utils import appdirs
  File "/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/pip/_internal/utils/appdirs.py", line 13, in <module>
    from pip._vendor import appdirs as _appdirs
ImportError: cannot import name \'appdirs\' from \'pip._vendor\' (/opt/venvs/securedrop-app-code/lib/python3.8/site-packages/pip/_vendor/__init__.py)

```

The same error can be observed by activating the virtualenv on the app server and running any pip command.

## Comments
Some potential resolutions already discussed:
- use dh_virtualenv's `--upgrade-pip-to` argument to install a working pip (e.g. 21.1.1) - this works but doesn't check hashes and adds a massive unreviewed dependency
- add pip to securedrop-app-code requirements - unclear if this will work, good for hashes, would still be on the hook to review
- figure out how to not install pip - by far the best solution but not sure it's possible
- live with the busted pip addition, drop the test - if pip is in fact required in the venv, this will backfire, plus test coverage reduced
- live with the busted pip addrion, update the test to check versions with importlib instead - same backfire potential but at least test number doesn't :arrow_down:.

**Comment by eloquence** [comment]
This was for now addressed via #5939. What additional follow-up is required here prior to the release of 2.0.0?

**Comment by legoktm** [comment]
I don't think there's anything left here to do.
